pkg/drift/assets/assets.go

```diff
@@ -0,0 +1,133 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package assets
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	asset "cloud.google.com/go/asset/apiv1"
+	"cloud.google.com/go/asset/apiv1/assetpb"
+	"google.golang.org/api/iterator"
+)
+
+const (
+	ORGANIZATION = "Organization"
+	FOLDER       = "Folder"
+	PROJECT      = "Project"
+)
+
+type HierarchyNode struct {
+	ID         int64
+	Name       string
+	ParentID   int64
+	ParentType string // Either Folder or Organization
+	NodeType   string // Either Folder or Organization or Project
+}
+
+type Client struct {
+	assetClient *asset.Client
+}
+
+func NewClient(ctx context.Context) (*Client, error) {
+	client, err := asset.NewClient(ctx)
+	if err != nil {
+		log.Fatalf("asset.NewClient: %v", err)
+	}
+
+	return &Client{
+		assetClient: client,
+	}, nil
+}
+
+func (c *Client) GetBuckets(ctx context.Context, organizationID int64, query string) ([]string, error) {
+	// gcloud asset search-all-resources \
+	// --asset-types=storage.googleapis.com/Bucket --query="$TERRAFORM_GCS_BUCKET_LABEL" --read-mask=name \
+	// "--scope=organizations/$ORGANIZATION_ID"
+	// var readMask fmpb.FieldMask
+	// readMask.Paths = append(readMask.Paths, "name")
+	req := &assetpb.SearchAllResourcesRequest{
+		Scope:      fmt.Sprintf("organizations/%d", organizationID),
+		AssetTypes: []string{"storage.googleapis.com/Bucket"},
+		Query:      query,
+		// ReadMask:   &readMask,
+	}
+	it := c.assetClient.SearchAllResources(ctx, req)
+	results := []string{}
+	for {
+		resource, err := it.Next()
+		if errors.Is(err, iterator.Done) {
+			break
+		}
+		if err != nil {
+			return nil, fmt.Errorf("error while iterating over assets %w", err)
+		}
+		results = append(results, strings.Replace(resource.Name, "//storage.googleapis.com/", "", 1))
+	}
+	return results, nil
+}
+
+func (c *Client) GetHierarchyAssets(ctx context.Context, organizationID int64, assetType string) ([]HierarchyNode, error) {
+	f := []HierarchyNode{}
+	req := &assetpb.SearchAllResourcesRequest{
+		Scope:      fmt.Sprintf("organizations/%d", organizationID),
+		AssetTypes: []string{assetType},
+	}
+	it := c.assetClient.SearchAllResources(ctx, req)
+	for {
+		resource, err := it.Next()
+		if errors.Is(err, iterator.Done) {
+			break
+		}
+		if err != nil {
+			return nil, fmt.Errorf("error while iterating over assets %w", err)
+		}
+		var id int64
+		assetType := strings.Replace(resource.AssetType, "cloudresourcemanager.googleapis.com/", "", 1)
+		if assetType == FOLDER {
+			id, err = strconv.ParseInt(strings.Replace(resource.Folders[0], "folders/", "", 1), 10, 64)
```

Comment:
It feels like `strings.TrimPrefix` would be more efficient and more explicit for these.